### PR TITLE
Populating pipelinrun attestation

### PR DESCRIPTION
**Pull request by bcaton85**
Populates the `pipelineRun` attestation with pipeline information. The approach being to have each directory under `pkg/chains/formats/intotoite6` have it's own `generateAttestation` method which will be called by `pkg/chains/formats/intotoite6/intotoite6.go`.

Note in this implementation materials are generated by looking for the `CHAINS-GIT_COMMIT` and `CHAINS-GIT_URL` fields in the parameters and results of the `pipelineRun`, similar to `taskRuns`.

Currently we're pulling all of this information out of the `pipelineRun` object itself since `taskRun` status information is stored directly into the status of the `pipelineRun` object. This will be disabled by default in a future release, instead links will be provided to the `taskRun` objects where we will need to make a new request for the object. [Here is a link to the documentation on this change.](https://tekton.dev/docs/pipelines/pipelineruns/#configuring-usage-of-taskrun-and-run-embedded-statuses)

Here as an example attestation. (Certain steps have been removed for brevity.)
```
_type: https://in-toto.io/Statement/v0.1
predicateType: https://slsa.dev/provenance/v0.2
subject:
  - name: quay.io/bcaton/contract
    digest:
      sha256: d7ec64d71b30134c150f1620026c2b7c6da3f1c4dbde318fb0391f50ecfd2d5d
predicate:
  builder:
    id: https://tekton.dev/chains/v2
  buildType: https://tekton.dev/attestations/chains@v2
  invocation:
    configSource: {}
    parameters:
      IMAGE: quay.io/bcaton/contract
      PUSH_SECRET_NAME: builder
  buildConfig:
    tasks:
      - name: git-clone
        startedAt: "2022-05-06T15:41:50Z"
        finishedAt: "2022-05-06T15:41:58Z"
        status: Succeeded
        steps:
          - stepState:
              terminated:
                exitCode: 0
                reason: Completed
                message: '[{"key":"commit","value":"d7be1eb9c26e7b45d58ac02520175474bc0657fc","type":1},{"key":"url","value":"https://github.com/caugello/single-nodejs-app","type":1}]'
                startedAt: "2022-05-06T15:41:57Z"
                finishedAt: "2022-05-06T15:41:58Z"
                containerID: cri-o://93f9d4e24e033f63b2e148cf3e7663e0ba351482240d2f4071e6bc6ba058c19e
              name: clone
              container: step-clone
              imageID: registry.redhat.io/openshift-pipelines/pipelines-git-init-rhel8@sha256:af7dd5b3b1598a980f17d5f5d3d8a4b11ab4f5184677f7f17ad302baa36bd3c1
            script: |
              #!/usr/bin/env sh
              git clone https://github.com/caugello/single-nodejs-app
      - name: build
        startedAt: "2022-05-06T15:41:59Z"
        finishedAt: "2022-05-06T15:46:45Z"
        status: Succeeded
        steps:
          - stepState:
              terminated:
                exitCode: 0
                reason: Completed
                startedAt: "2022-05-06T15:42:11Z"
                finishedAt: "2022-05-06T15:42:12Z"
                containerID: cri-o://288c4a889a7da19fb6b72787d31149d84a5eb96fa903a0ab141a407cf8f57c4e
              name: generate
              container: step-generate
              imageID: registry.redhat.io/ocp-tools-4-tech-preview/source-to-image-rhel8@sha256:cd4996fba88519ec21499da63d8c3e26cc4552429b949da76914d0666c27872d
            command:
              - s2i
              - build
              - $(params.PATH_CONTEXT)
              - image-registry.openshift-image-registry.svc:5000/openshift/nodejs:$(params.VERSION)
              - --as-dockerfile
              - /gen-source/Dockerfile.gen
  metadata:
    buildStartedOn: "2022-05-06T15:41:50Z"
    buildFinishedOn: "2022-05-06T15:46:45Z"
    completeness:
      parameters: false
      environment: false
      materials: false
    reproducible: false
  materials:
    - uri: git+https://github.com/caugello/single-nodejs-app.git
      digest:
        sha1: d7be1eb9c26e7b45d58ac02520175474bc0657fc
```

### CODE REVIEW

File: pkg/chains/formats/intotoite6/util/utils.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// It also goes through looking for any PipelineResources of Image type
+func GetSubjectDigests(obj objects.K8sObject, logger *zap.SugaredLogger) []intoto.Subject {
+	var subjects []intoto.Subject
+	imgs := artifacts.ExtractOCIImagesFromResults(obj, logger)
+	for _, i := range imgs {
+		if d, ok := i.(name.Digest); ok {
+			subjects = append(subjects, intoto.Subject{
+				Name: d.Repository.Name(),
+				Digest: slsa.DigestSet{
+					"sha256": strings.TrimPrefix(d.DigestStr(), "sha256:"),
```
Comment (zregvart) [inline pkg/chains/formats/intotoite6/util/utils.go:36]: Nit: perhaps split `d.DigestStr()` on `:`? Might be far ahead but one less thing to worry about when switching to `sha512` or other

**Review by zregvart — APPROVED** [review]
LGTM

File: pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+		},
+	}
+	return att, nil
+}
+
+func invocation(pr *v1beta1.PipelineRun) slsa.ProvenanceInvocation {
+	i := slsa.ProvenanceInvocation{}
+	params := make(map[string]string)
+	// add params
+	if ts := pr.Status.PipelineSpec; ts != nil {
```
Comment (lcarva) [inline pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go:67]: Nitpick: `s/ts/ps`

File: pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+	// pipelineRun.status.taskRuns doesn't maintain order,
+	// so we'll store here and use the order from pipelineRun.status.pipelineSpec.tasks
+	taskRuns := make(map[string]*v1beta1.PipelineRunTaskRunStatus)
+	for _, tr := range pr.Status.TaskRuns {
+		taskRuns[tr.PipelineTaskName] = tr
+	}
+
+	for _, tr := range pr.Status.PipelineSpec.Tasks {
+		trStatus := taskRuns[tr.Name]
```
Comment (lcarva) [inline pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go:101]: Nitpick: maybe the `taskRuns` variable name should include `status` in some form to better describe it, e.g. `trStatuses`

File: pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+				commit = p.Default.StringVal
+				continue
+			}
+			if p.Name == util.UrlParam {
+				url = p.Default.StringVal
+			}
+		}
+	}
+
+	// search status.TaskRunResults
```
Comment (lcarva) [inline pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go:170]: `s/TaskRunResults/PiplineResults`

File: pkg/chains/formats/intotoite6/util/utils.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
+package util
```
Comment (lcarva) [inline pkg/chains/formats/intotoite6/util/utils.go:1]: Would it make sense to just move the contents of this file directly into `intotoite6.go`?

File: pkg/config/store_test.go [hunk 6]
```diff
@@ -114,7 +114,7 @@ func TestParse(t *testing.T) {
 						Signer:         "x509",
 					},
 					PipelineRuns: Artifact{
-						Format:         "tekton-pipeline-run",
```
Comment (lcarva) [inline pkg/config/store_test.go:117]: Thanks for fixing this :)

File: pkg/chains/formats/intotoite6/util/utils.go (hunk 5 above)
Comment (bcaton85) [reply pkg/chains/formats/intotoite6/util/utils.go:1]: This causes a cyclical import - intotoite6 -> pipelinerun, pipelinerun -> intotoite6

File: pkg/config/store_test.go (hunk 6 above)
Comment (bcaton85) [reply pkg/config/store_test.go:117]: No problem!

File: pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go (hunk 3 above)
Comment (bcaton85) [reply pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go]: yeah makes sense, updated

File: pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go (hunk 4 above)
Comment (bcaton85) [reply pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go]: updated

File: pkg/chains/formats/intotoite6/util/utils.go (hunk 1 above)
Comment (lcarva) [reply pkg/chains/formats/intotoite6/util/utils.go]: This is weird now:
```
"sha256": strings.Split(d.DigestStr(), ":")[1],
```

Comment (bcaton85) [reply pkg/chains/formats/intotoite6/util/utils.go]: Yeah, maybe we keep `strings.TrimPrefix(d.DigestStr(), "sha256:")` for now. I'd rather keep the original implementation intact as possible.

**Pull request merged**